ambry-server/src/main/java/com.github.ambry.server/AmbryRequests.java

```diff
@@ -816,20 +849,21 @@ private AdminResponse handleBlobStoreControlRequest(DataInputStream requestStrea
               }
             } else {
               error = ServerErrorCode.Unknown_Error;
-              logger.error("Disable compaction on given BlobStore failed for {}", partitionId);
+              logger.error("Disable compaction fails on given BlobStore {}", partitionId);
             }
+          } else {
+            error = ServerErrorCode.Bad_Request;
+            logger.debug("The number of replicas to catch up should not be less or equal to zero {}",
+                blobStoreControlAdminRequest.getNumReplicasCaughtUpPerPartition());
           }
         } else {
-          logger.debug("Validate request fails for {} with error code {}", partitionId, error);
+          logger.debug("Validate request fails for {} with error code {} when trying to stop store", partitionId,
+              error);
         }
-      } else {
-        error = ServerErrorCode.Partition_Unknown;
-        logger.debug("The partition Id should not be null.");
       }
     } else {
-      error = ServerErrorCode.Bad_Request;
-      logger.debug("The number of replicas to catch up should not be less or equal to zero {}",
-          blobStoreControlAdminRequest.getNumReplicasCaughtUpPerPartition());
+      error = ServerErrorCode.Partition_Unknown;
```

Comment:
Absence of a partition altogether should remain as a bad request. Partition unknown really means that there is a partition in the request that is not recognized by this node.